aten/src/ATen/Tensor.cpp

```diff
@@ -12,4 +12,8 @@ void Tensor::print() const {
   }
 }
 
+bool Tensor::isVariable() const noexcept {
```

Comment:
inline in a header please

Reply:
I couldn't right now because Tensor.h doesn't include Type.h. Should I  include it?

Reply:
It seemed like there was a reason it wasn't included before, but not sure.

Reply:
`Type.h` includes `Tensor.h`, this isn't gonna work, it would be recursive

Reply:
I remember there being some include order problems that would otherwise force a huge amount of duplicated declarations. You can still just put Tensor::isVariable into `Type.h`, though. See if we do this or an equivalent for any other Tensor methods already.

Reply:
That said, `Type.h` doesn't need to include `Tensor.h` it looks like

Reply:
Ok yeah that's what Ed had. It just seemed weird